### PR TITLE
RS/YJ/Rule 11-17 (include only necessary codes) 

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
This PR replaces PR #1546. Thank you!

### CODE REVIEW

File: rct229/rulesets/ashrae9012019/section11/section11rule17.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+                swh_bat = data["swh_bat"]
+                service_water_heating_use_ids = data["service_water_heating_use_ids"]
+
+                return {
+                    "swh_bat": swh_bat[bldg_seg_id_p],
+                    "service_water_heating_use_ids": service_water_heating_use_ids[
+                        bldg_seg_id_p
+                    ],
+                }
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule17.py:93]: Why do we need a `create_data` in the `RuleDefinitionBase` class?
We could just move the above logic to `get_calc_vals`

File: rct229/rulesets/ashrae9012019/section11/section11rule17.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                        getattr_(swh_use, "service_water_heating_uses", "use") > 0.0
+                        for service_water_heating_use_id in service_water_heating_use_ids
+                        for swh_use in find_all(
+                            f'$.zones[*].spaces[*].service_water_heating_uses[*][?(@.id="{service_water_heating_use_id}")]',
+                            swh_use_p,
+                        )
+                    ]
+                )
+
+                return {"has_swh_loads": has_swh_loads}
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule17.py:110]: In return, including `swh_bat`

**Review by weilixu — APPROVED** [review]
There are no major logic issue but prefer doing all logic under `get_calc_vals` and in the reporting dictionary, include the `swh_bat`

File: rct229/rulesets/ashrae9012019/section11/section11rule17.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule17.py:110]: Addressed.

File: rct229/rulesets/ashrae9012019/section11/section11rule17.py (hunk 1 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule17.py:93]: Good point. Addressed.

**Pull request merged**